firebase-app-distribution/src/main/java/com/google/firebase/app/distribution/FirebaseAppDistribution.java

```diff
@@ -132,41 +135,124 @@ public UpdateTask updateIfNewReleaseAvailable() {
 
       cachedUpdateIfNewReleaseTask = new UpdateTaskImpl();
     }
-    checkForNewRelease()
-        .onSuccessTask(
-            release -> {
-              if (release == null) {
-                postProgressToCachedUpdateIfNewReleaseTask(
-                    UpdateProgress.builder()
-                        .setApkFileTotalBytes(UNKNOWN_RELEASE_FILE_SIZE)
-                        .setApkBytesDownloaded(UNKNOWN_RELEASE_FILE_SIZE)
-                        .setUpdateStatus(UpdateStatus.NEW_RELEASE_NOT_AVAILABLE)
-                        .build());
-                setCachedUpdateIfNewReleaseResult();
-                return Tasks.forResult(null);
+
+    if (!isTesterSignedIn()) {
+      Activity currentActivity = lifecycleNotifier.getCurrentActivity();
+      if (currentActivity == null) {
+        LogWrapper.getInstance().e("No foreground activity found.");
+        return getErrorUpdateTask(
+            new FirebaseAppDistributionException(
+                ErrorMessages.APP_BACKGROUNDED,
+                FirebaseAppDistributionException.Status.UPDATE_NOT_AVAILABLE));
+      }
+
+      signInDialog =
+          getSignInDialog(
+              currentActivity,
+              firebaseApp,
+              (dialogInterface, i) ->
+                  checkForNewRelease() // check for new release will call sign in tester
```

Comment:
This is a great question, and something we've gone back and forth with. 

We chose to have checkForNewRelease call signInTester to keep the methods aligned with the iOS SDK. Granted, the SDKs already deviate, so we could remove signIn from the SDK and be okay with the different behaviors, but this may impact Flutter and other cross-platform frameworks